### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by george827**
Hello, in this feature branch I,

### **Highlights,**

- [ ] Created a class `Classroom`
- [ ] Created the _has-many/belongs-to_ relationship between `Classroom` and `Student.`
- [ ] Created a class `Book`.
- [ ] Created a class `Rental`.
- [ ] Created the _many-to-many/ has-many-through_ relationship between `Person` and `Book` using the intermediate class `Rental`

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (DammyShittu) [inline book.rb]: - [ ] Kindly add `:rentals` to attr_accessor instead of attr_reader

File: person.rb [hunk 2]
```diff
@@ -1,9 +1,26 @@
-class Person
+require './naeable'
+require './capitalizedecorator'
+require './trimmerdecorator'
+
+class Person < Nameable
+  attr_reader :id, :rentals
+  attr_writer :name, :age
```
Comment (DammyShittu) [inline person.rb:8]: - [ ] We want to be able to read and write to `:rentals`. Kindly set this up.

  ```
  attr_accessor :rentals
  ```

File: person.rb [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
+
```
Comment (DammyShittu) [inline person.rb]: - [ ] We want to allow a person to create a rental given a date and a book object. That is the purpose of this method. Kindly refactor the method:

  ```
  def add_rental(date, book)
    Rental.new(date, book, self)
  end
  ```

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (DammyShittu) [inline book.rb]: - [ ] We want to allow a person to create a rental given a date and a person object. That is the purpose of this method. Kindly refactor the method:

  ```
  def add_rental(date, person)
    Rental.new(date, self, person)
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @george827,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @george827,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Good pull request message ✔️
- Well organized and clean project ✔️
approval1


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 3 above)
Comment (george827) [reply person.rb]: @DammyShittu and how will it add the rental to the rentals array?
and what does `self`  do , in rental I have (date, book, person) as params
[rental-code](https://github.com/george827/school_library/blob/24f9ce0b2b0fbd7fd7a381be9f46c88b5c854079/rental.rb#LL4C3-L13C4)

**Pull request merged**